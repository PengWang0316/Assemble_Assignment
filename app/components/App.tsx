import React, { Suspense, lazy, ReactElement } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
// import importedComponent from 'react-imported-component';
import CssBaseline from '@material-ui/core/CssBaseline';

import Navbar from './Navbar';
import { HOME_PAGE_URL, BILL_PAGE_URL, COMPLETE_PAGE_URL } from '../config';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#4081f5',
      main: '#1a4fad',
      dark: '#08275e',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
// const HomePage = importedComponent(() => import(/* webpackChunkName: "HomePageContainer" *//* webpackPrefetch: true */ './containers/HomePageContainer').catch(err => console.log(err)));
const CartPage = lazy(() => import(/* webpackChunkName: "CartPageContainer" *//* webpackPrefetch: true */ './containers/Cart'));
const BillPage = lazy(() => import(/* webpackChunkName: "BillContainer" *//* webpackPrefetch: true */ './containers/BillInformation'));
const OrderCompletePage = lazy(() => import(/* webpackChunkName: "OrderCompletePageContainer" *//* webpackPrefetch: true */ './containers/OrderComplete'));


/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export const App = (): ReactElement => (
  <MuiThemeProvider theme={theme}>
    <Router>
      <div>
        <CssBaseline />
        <Navbar />
        <main>
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route exact path={HOME_PAGE_URL} component={CartPage} />
              <Route exact path={BILL_PAGE_URL} component={BillPage} />
              <Route exact path={COMPLETE_PAGE_URL} component={OrderCompletePage} />
              <Route render={() => <p>Not Fount!</p>} />
            </Switch>
          </Suspense>
        </main>
      </div>
    </Router>
  </MuiThemeProvider>
);
export default App;
