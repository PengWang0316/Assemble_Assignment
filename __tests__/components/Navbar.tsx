import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';

import { Navbar } from '../../app/components/Navbar';

jest.mock('react-router-dom', () => ({ Link: 'Link', withRouter: jest.fn() }));


describe('Test the Navbar component', () => {
  let component: ShallowWrapper;
  beforeAll(() => {
    component = shallow(<Navbar />);
  });

  test('should have the text "Assemble Store"', () => {
    expect(component.text()).toBe('Assemble Store');
  });

});
