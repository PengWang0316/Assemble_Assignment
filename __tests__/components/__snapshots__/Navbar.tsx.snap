// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Navbar component should match the snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic makeStyles-appbar-4 MuiAppBar-colorPrimary"
  data-testid="navbar"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <Link
      className="makeStyles-link-1 makeStyles-flex1-3"
      data-testid="titleLink"
      to="/"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
      >
        Assemble Store
      </h6>
    </Link>
  </div>
</header>
`;
